swarm/src/behaviour/either.rs

```diff
@@ -49,173 +47,42 @@ where
         }
     }
 
-    fn inject_connection_established(
-        &mut self,
-        peer_id: &PeerId,
-        connection: &ConnectionId,
-        endpoint: &ConnectedPoint,
-        errors: Option<&Vec<Multiaddr>>,
-        other_established: usize,
-    ) {
+    fn on_swarm_event(&mut self, event: behaviour::FromSwarm<Self::ConnectionHandler>) {
         match self {
-            Either::Left(a) => a.inject_connection_established(
-                peer_id,
-                connection,
-                endpoint,
-                errors,
-                other_established,
-            ),
-            Either::Right(b) => b.inject_connection_established(
-                peer_id,
-                connection,
-                endpoint,
-                errors,
-                other_established,
-            ),
-        }
-    }
-
-    fn inject_connection_closed(
-        &mut self,
-        peer_id: &PeerId,
-        connection: &ConnectionId,
-        endpoint: &ConnectedPoint,
-        handler: <Self::ConnectionHandler as IntoConnectionHandler>::Handler,
-        remaining_established: usize,
-    ) {
-        match (self, handler) {
-            (Either::Left(behaviour), Either::Left(handler)) => behaviour.inject_connection_closed(
-                peer_id,
-                connection,
-                endpoint,
-                handler,
-                remaining_established,
-            ),
-            (Either::Right(behaviour), Either::Right(handler)) => behaviour
-                .inject_connection_closed(
-                    peer_id,
-                    connection,
-                    endpoint,
-                    handler,
-                    remaining_established,
-                ),
-            _ => unreachable!(),
+            Either::Left(b) => b.on_swarm_event(event.map_handler(
```

Comment:
@thomaseizinger comment on https://github.com/libp2p/rust-libp2p/pull/3085#discussion_r1015994812 sparked a thought.

What if `b` implements the `inject_*` style methods and not the `on_*` style methods? In such case, this call would be lost, right?

Unless I am missing something combinators like `Either` need to call the `inject_*` style methods until we remove them.

Reply:
No no, I noticed it when replacing `Inject_*` on combinators for #3085 but with the deprecation/removal discussion for to address that here, thanks for reminding @mxinden! Updated to call `inject` methods on those structs. Also merged from master and fixed conflicts ptal